go/private/rules/binary.bzl

```diff
@@ -118,23 +114,34 @@ def _go_tool_binary_impl(ctx):
     name = ctx.label.name
     if sdk.goos == "windows":
         name += ".exe"
-    out = ctx.actions.declare_file(name)
 
-    command_tpl = ("{go} tool compile -o {out}.a -I {goroot} -trimpath=$PWD $@ && " +
-                   "{go} tool link -o {out} -L {goroot} {out}.a && " +
-                   "rm {out}.a")
-    command = command_tpl.format(
-        go = shell.quote(sdk.go.path),
-        goroot = shell.quote(sdk.root_file.dirname),
-        out = shell.quote(out.path),
+    cout = ctx.actions.declare_file(name + ".a")
+    cargs = ctx.actions.args()
+    cargs.add_all(["tool", "compile"])
+    cargs.add("-o", cout)
+    cargs.add("-I", sdk.root_file.dirname)
+    cargs.add("-trimpath=.")
```

Comment:
This won't work: the compiler treats these paths literally. The path has to be the absolute path to the directory containing the source files, but there's no way during analysis to know what that will be.

I think this means some kind of shell script is still needed. Perhaps on Windows it could be a .bat file instead?

Reply:
Could we fix this upstream instead? It seems pretty reasonable to allow relative paths.

Reply:
Are we sure trimpath doesn't work actually? How does this work internally? Are all Go builds surrounded by such a wrapper script? You can't put absolute paths into the command line or it won't be cacheable when distributed.